pkg/models/datasource_cache.go

```diff
@@ -55,14 +57,37 @@ func (ds *DataSource) GetHttpTransport() (*http.Transport, error) {
 
 	tlsConfig.Renegotiation = tls.RenegotiateFreelyAsClient
 
-	transport := &http.Transport{
-		TLSClientConfig: tlsConfig,
-		Proxy:           http.ProxyFromEnvironment,
-		Dial: (&net.Dialer{
+	var proxy func(*http.Request) (*url.URL, error)
+	var dial func(network, address string) (net.Conn, error)
+
+	switch strings.Split(ds.Url, ":")[0] {
```

Comment:
https://golang.org/src/net/url/url.go?s=13181:13220#L469 
and getting the schema from *URL would be more elegant IMO.

Reply:
updated. thanks @bergquist